xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp

```diff
@@ -78,6 +78,8 @@ bool CRendererDRMPRIME::Configure(const VideoPicture& picture, float fps, unsign
 void CRendererDRMPRIME::AddVideoPicture(const VideoPicture& picture, int index, double currentClock)
 {
   BUFFER& buf = m_buffers[index];
+  if (buf.videoBuffer)
```

Comment:
you should log an error here. LinuxRendererGL (reference) does not have this check. Means if it ever releases a buffer here, something went really wrong.

Reply:
this is a workaround and we expect to release video buffers here after a call has been made to `Flush`, if a video buffer is released and removed in `Flush` while it is tied to the primary drm plane the entire drm display pipeline can be disabled.

a proper solution where the video buffers is reference counted while being displayed on screen (tied to a drm plane) will follow later, may require some refactoring and hooks into gbm windowing

Reply:
maybe place a comment here why you do this

Reply:
a short comment has been added